### PR TITLE
TF2 porting: Date Feature and Encoders

### PR DESCRIPTION
**Pull request by jimthompson5802**
# Code Pull Requests
Port of the Date encoders and feature.  One item to mention. 
* I had to modify some of the tf dtype to get it to work.  

This change passes the `test_experiment.py::test_experiment_date` unit test.
``` 
ytest -v test_experiment.py::test_experiment_date
======================================== test session starts ========================================
platform linux -- Python 3.6.9, pytest-5.4.3, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /opt/project
plugins: pycharm-0.6.0, typeguard-2.8.0
collected 1 item

test_experiment.py::test_experiment_date PASSED                                               [100%]

========================================= warnings summary ==========================================
/usr/local/lib/python3.6/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py:15
  /usr/local/lib/python3.6/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py:15: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

tests/integration_tests/test_experiment.py::test_experiment_date
tests/integration_tests/test_experiment.py::test_experiment_date
  /usr/local/lib/python3.6/dist-packages/sklearn/metrics/_classification.py:1221: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
    _warn_prf(average, modifier, msg_start, len(result))

tests/integration_tests/test_experiment.py::test_experiment_date
  /usr/local/lib/python3.6/dist-packages/sklearn/metrics/_classification.py:1221: UndefinedMetricWarning: Recall is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
    _warn_prf(average, modifier, msg_start, len(result))

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=================================== 1 passed, 4 warnings in 3.38s ==================================
```

### CODE REVIEW

File: ludwig/features/date_feature.py [hunk 1]
```diff
@@ -115,56 +116,27 @@ def add_feature_data(
 
 
 class DateInputFeature(DateBaseFeature, InputFeature):
-    def __init__(self, feature):
-        super().__init__(feature)
+    def __init__(self, feature, encoder_obj=None):
+        DateBaseFeature.__init__(self, feature)
+        InputFeature.__init__(self)
 
         self.encoder = 'embed'
```
Comment (w4nderlust) [inline ludwig/features/date_feature.py:123]: this should be a class variable

**Comment by w4nderlust** [comment]
Added a minor comment, otherwise it looks good to me. I only have one doubt, but that's a problem that was there in the original implementation too:
`embedded_weekdays` on 236 does:
`input_vector[:, 3]` without the -1. Wanted to be sue that the values are between 0 and 6, and not 1 and 7, if they are between 1 and 7 we should do `input_vector[:, 3] - 1`

**Comment by jimthompson5802** [comment]
Incorporated your comment re: class attribute

re:
> Wanted to be sue that the values are between 0 and 6,

I generated some data that spanned Monday through Sunday, confirmed the generated values in range of [0, 6]

**Comment by w4nderlust** [comment]
> Incorporated your comment re: class attribute
> 
> re:
> 
> > Wanted to be sue that the values are between 0 and 6,
> 
> I generated some data that spanned Monday through Sunday, confirmed the generated values in range of [0, 6]

great so the original implementation was correct to begin with :) merging this.

**Pull request merged**